listener.go

```diff
@@ -22,6 +22,8 @@ type hub struct {
 	// Downstream service state
 	downstreamState baps3.ServiceState
 
+	Autoadvance bool
```

Comment:
should probably work out how to be consistent with variable names inside the same package, but for now, i'd recommend `autoAdvance` for consistency with existing stuff